cmd/argo/commands/root.go

```diff
@@ -128,10 +132,35 @@ If your server is behind an ingress with a path (you'll be running "argo server
 		cli.SetLogLevel(logLevel)
 		cmdutil.SetGLogLevel(glogLevel)
 		log.WithField("version", argo.GetVersion()).Debug("CLI version")
+		printVersionMismatchWarning(command)
 	}
 	command.PersistentFlags().StringVar(&logLevel, "loglevel", "info", "Set the logging level. One of: debug|info|warn|error")
 	command.PersistentFlags().IntVar(&glogLevel, "gloglevel", 0, "Set the glog logging level")
 	command.PersistentFlags().BoolVarP(&verbose, "verbose", "v", false, "Enabled verbose logging, i.e. --loglevel debug")
 
 	return command
 }
+
+// printVersionMismatchWarning logs a warning if the CLI version does not match the server version
+func printVersionMismatchWarning(command *cobra.Command) {
+	// if ARGO_SERVER isn't set there's no need to compare server and cli version
+	if _, ok := os.LookupEnv("ARGO_SERVER"); !ok {
+		return
+	}
+	ctx, apiClient := client.NewAPIClient(command.Context())
+	serviceClient, err := apiClient.NewInfoServiceClient()
+	// fail with warning because check for version mismatch should not prevent other commands from running
+
+	if err != nil {
+		log.Warnf("Failed create service client: %v", err)
+		return
+	}
+	serverVersion, err := serviceClient.GetVersion(ctx, &infopkg.GetVersionRequest{})
+	if err != nil {
+		log.Warnf("Failed to connect to Argo Server: %v", err)
+		return
+	}
+	if serverVersion.GitTag != argo.GetVersion().GitTag {
+		log.Warnf("CLI version (%s) does not match server version (%s). This can lead to unexpected behavior.", argo.GetVersion().GitTag, serverVersion.GitTag)
+	}
```

Comment:
I think it might be a better idea if you separated out the logic that checks for a mismatch from the code that presents this information to the user. 

Only saying this because we could make this a bit more strict then, we could fail on a version mismatch and have a flag to force any action to be performed regardless.

Reply:
ah sorry for not having able to continue working on this PR. work and private life got quite busy as of late. 
I'm not sure this will change a lot in the coming months, but I do think that's a great idea @isubasinghe

Reply:
No worries, take your time!